src/nvim/window.c

```diff
@@ -4745,7 +4745,9 @@ void win_new_width(win_T *wp, int width)
   wp->w_redr_status = TRUE;
 
   if (wp->w_buffer->terminal) {
-    terminal_resize(wp->w_buffer->terminal, wp->w_width, 0);
+    if (wp->w_height != 0) {
+      terminal_resize(wp->w_buffer->terminal, wp->w_width, 0);
```

Comment:
Checking `wp->w_height` seems to add some technical debt. We're baking-in knowledge of `terminal_resize `'s implementation (we're passing the literal `0` yet we're checking `wp->w_height` for non-zero... that makes no sense). I suggest removing this check in your other PR.